core/src/main/scala/org/apache/spark/scheduler/CoarseCookSchedulerBackend.scala

```diff
@@ -251,89 +275,89 @@ class CoarseCookSchedulerBackend(
       s"[ -z $$KEEP_SPARK_LOCAL_TARS ] || rm -f $$(basename $uri)"
     }
 
-    val cmds =
-      debug ++
-        environment ++
-        urisCommand ++
+    val commandSeq =
+      debugCommand ++
+        envSettingCommand ++
+        uriFetchCommand ++
         shippedTarballsCommand ++
-        remoteConfFetch ++
+        remoteConfFetchCommand ++
         keystorePull.map(Seq(_)).getOrElse(Seq[String]()) ++
         Seq("set", commandString) ++
         cleanup
 
     val builder = new Job.Builder()
-      .setUUID(jobId)
-      .setName(schedulerConf.getPrefixOfCookJobName)
-      .setCommand(cmds.mkString("; "))
+      .setUUID(jobUUID)
+      .setName(schedulerConf.cookJobNamePrefix)
+      .setCommand(commandSeq.mkString("; "))
       .setMemory(executorMemory(sc).toDouble)
-      .setCpus(numCores)
-      .setPriority(schedulerConf.getPriorityPerCookJob)
+      .setCpus(cores.toDouble)
+      .setPriority(schedulerConf.cookJobPriority)
+      // The following two setting ensure each Cook job has only 1 instance.
       .disableMeaCulpaRetries()
       .setRetries(1)
 
-    val container = conf.get("spark.executor.cook.container", null)
-    if (container != null) {
-      builder.setContainer(new JSONObject(container))
+    conf.getOption(schedulerConf.SPARK_EXECUTOR_COOK_CONTAINER).foreach {
```

Comment:
To be clear, I think CookSchedulerConfiguration is good. What I'm wondering is why we went from having `schedulerConf.dynamicAllocationEnabled` to having `conf.getOption(schedulerConf.DYNAMIC_ALLOCATION_ENABLED)`. I think it's good to just put those methods in scheduler conf and leave them there.